react/jsx-curly-brace-presence "children" "always" complains about whitespace

```
"eslint": "^4.18.2",
"eslint-plugin-react": "^7.6.1",
```

I have the prop: `"react/jsx-curly-brace-presence":["error", { "props": "always", "children": "always" }],` in my `.eslintrc` file because I'm trying to convert:

```jsx
<div className="NameCapture_wrapper">
    <p className="NameCapture__question">
        What's your first name?
    </p>
    <NameCaptureNameInput/>
</div>
```
to:
```jsx
<div className={`NameCapture_wrapper`}>
    <p className={`NameCapture__question`}>
        {`What's your first name?`}
    </p>
    <NameCaptureNameInput/>
</div>
```
but eslint is complaining about the whitespace:

```
  10:44  error  Need to wrap this literal in a JSX expression          react/jsx-curly-brace-presence
  13:49  error  Need to wrap this literal in a JSX expression          react/jsx-curly-brace-presence
  14:56  error  Need to wrap this literal in a JSX expression          react/jsx-curly-brace-presence
  17:53  error  Need to wrap this literal in a JSX expression          react/jsx-curly-brace-presence
  18:45  error  Need to wrap this literal in a JSX expression          react/jsx-curly-brace-presence
  19:23  error  Need to wrap this literal in a JSX expression          react/jsx-curly-brace-presence
```

Below is what fixes it but I shouldn't have to put all of my JSX on one line right?
```jsx
<div className={`NameCapture_wrapper`}><p className={`NameCapture__question`}>{`What's your first name?`}</p><NameCaptureNameInput/></div>
```

--- a/lib/rules/jsx-curly-brace-presence.js
+++ b/lib/rules/jsx-curly-brace-presence.js
@@ -298,7 +298,11 @@ module.exports = {
       return areRuleConditionsSatisfied(parent, config, OPTION_NEVER);
     }
 
-    function shouldCheckForMissingCurly(parent, config) {
+    function shouldCheckForMissingCurly(node, config) {
+      if (node.raw.trim() === '') {
+        return false;
+      }
+      const parent = node.parent;
       if (
         parent.children &&
         parent.children.length === 1 &&
@@ -322,7 +326,7 @@ module.exports = {
       },
 
       'Literal, JSXText': (node) => {
-        if (shouldCheckForMissingCurly(node.parent, userConfig)) {
+        if (shouldCheckForMissingCurly(node, userConfig)) {
           reportMissingCurly(node);
         }
       }

--- a/tests/lib/rules/jsx-curly-brace-presence.js
+++ b/tests/lib/rules/jsx-curly-brace-presence.js
@@ -376,6 +376,21 @@ ruleTester.run('jsx-curly-brace-presence', rule, {
         <MyComponent p={<Foo>Bar</Foo>}>
         </MyComponent>
       `
+    },
+    {
+      code: `
+        <MyComponent>
+          <div>
+            <p>
+              <span>
+                {"foo"}
+              </span>
+            </p>
+          </div>
+        </MyComponent>
+      `,
+      parser: parsers.BABEL_ESLINT,
+      options: [{children: 'always'}]
     }
   ],
 

